scripts/public_api_report.py

```diff
@@ -784,6 +854,394 @@ class CodeReferenceFilter:
+            name="self_check",
+            filepath=pathlib.Path(
+                "great_expectations/datasource/data_connector/runtime_data_connector.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="self_check is mentioned but in the docs we currently recommend using test_yaml_config which uses self_check under the hood. E.g. https://docs.greatexpectations.io/docs/guides/setup/configuring_data_contexts/how_to_configure_datacontext_components_using_test_yaml_config/#steps",
+            name="self_check",
+            filepath=pathlib.Path("great_expectations/datasource/new_datasource.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="False match for dict `.update()` method.",
+            name="update",
+            filepath=pathlib.Path(
+                "great_expectations/execution_engine/execution_engine.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="Included in Validator public api",
+            name="head",
+            filepath=pathlib.Path(
+                "great_expectations/execution_engine/sparkdf_execution_engine.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="Currently only used in testing code, not referenced in docs.",
+            name="close",
+            filepath=pathlib.Path(
+                "great_expectations/execution_engine/sqlalchemy_execution_engine.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="False match for Python `dict`",
+            name="dict",
+            filepath=pathlib.Path(
+                "great_expectations/render/renderer_configuration.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="False match for context.get_validator()",
+            name="get_validator",
+            filepath=pathlib.Path(
+                "great_expectations/rule_based_profiler/domain_builder/domain_builder.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="False match for context.get_validator()",
+            name="get_validator",
+            filepath=pathlib.Path(
+                "great_expectations/rule_based_profiler/helpers/util.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="False match for context.get_validator()",
+            name="get_validator",
+            filepath=pathlib.Path(
+                "great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="Not used directly but from Data Assistant or RuleBasedProfiler",
+            name="run",
+            filepath=pathlib.Path(
+                "great_expectations/rule_based_profiler/rule/rule.py"
+            ),
+        ),
```

Comment:
Are these needed for the Data Assistant contribution process?

Reply:
I don't see either `Rule.run()` or `ParameterBuilder.get_validator()` used in DataAssistant, OnboardingDataAssistant or the VolumeDataAssistant. It looks like it gets called internally. 
Could you point me to the Data Assistant contribution process to double check? I haven't been able to find it in our docs.